### PR TITLE
Adding Linux support for  PAC CLI actions

### PR DESCRIPTION
**Pull request by tehcrashxor**
- Adding the Linux version of the PAC CLI package
- A few pathing and OS check changes to permit the PAC runner to use the Linux executable
- Updated dist folder with the changes from the above

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -7,7 +7,11 @@ on:
 
 jobs:
   build-and-test:
-    runs-on: windows-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
```
Comment (davidjenni) [inline .github/workflows/pull-request.yml:11]: what is the plan to migrate the rolling-build workflow to linux as well and queue a manually triggered run on that one? I'm ok to make that as a separate PR if this unduly delays this PR

File: gulpfile.js [hunk 2]
```diff
@@ -160,6 +162,8 @@ async function createDist() {
 
 const recompile = gulp.series(
     clean,
+    async () => nugetInstall('CAP_ISVExp_Tools_Daily', 'Microsoft.PowerApps.CLI.Core.linux-x64', '1.5.6-daily-21031515', path.resolve(outdir, 'pac_linux')),
+    async () => { if (os.platform() === 'linux') { return fs.chmod(path.resolve(outdir, 'pac_linux', 'tools', 'pac'), 0o711) } },
```
Comment (davidjenni) [inline gulpfile.js:166]: nit: the os check shouldn't be necessary: node's fs.chmod() does "succeed" on windows as well

**Review by davidjenni — APPROVED** [review]

File: gulpfile.js (hunk 2 above)
Comment (tehcrashxor) [reply gulpfile.js]: Nice.  Updated.

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (tehcrashxor) [reply .github/workflows/pull-request.yml:11]: The changes for the rolling build are fairly simple, but as-is we'll need to set `max-parallel: 1` since the arguments are the same, and one OS's run could delete the test org while the other is still using it.

Looks like the rolling build is already failing on the main branch.  Should I go ahead and add the linux run as well, or wait until the rolling build is back in a good state?

Comment (davidjenni) [reply .github/workflows/pull-request.yml:11]: let's wait with adding linux to rolling until the win build is good. The error is weird, the envId doesn't resolve, but looking from e.g. my local pac admin list, it does under the given env guid (using latest pac CLI from master).

File: src/lib/exeRunner.ts [hunk 3]
```diff
@@ -51,19 +46,27 @@ export class ExeRunner {
...
+            process.stderr.on('data', (data) => stderr.push(...data.toString().split(os.EOL)));
 
-            pac.on('close', (code) => {
+            process.on('exit', (code) => {
                 if (code === 0) {
                     this.logger.info(`success: ${stdout.join(os.EOL)}`);
                     resolve(stdout);
+
+                    // Close out handles to the output streams so that we don't wait on grandchild processes like pacTelemetryUpload
+                    process.stdout.destroy();
```
Comment (davidjenni) [inline src/lib/exeRunner.ts:60]: move the common code to after the if/else block, since the handle closures are now the same

**Review by davidjenni — CHANGES_REQUESTED** [review]

File: src/lib/exeRunner.ts (hunk 3 above)
Comment (tehcrashxor) [reply src/lib/exeRunner.ts]: moved

**Review by davidjenni — APPROVED** [review]
